src/include/planner/operator/logical_filter.h

```diff
@@ -6,12 +6,21 @@
 namespace kuzu {
 namespace planner {
 
+struct LogicalFilterPrintInfo final : OPPrintInfo {
+    std::shared_ptr<binder::Expression> expression;
+    explicit LogicalFilterPrintInfo(std::shared_ptr<binder::Expression> expression)
+        : expression{std::move(expression)} {}
+    std::string toString() const override { return expression->toString(); }
+};
+
 class LogicalFilter final : public LogicalOperator {
 public:
     LogicalFilter(std::shared_ptr<binder::Expression> expression,
-        std::shared_ptr<LogicalOperator> child)
+        std::shared_ptr<LogicalOperator> child, common::cardinality_t cardinality = 0)
         : LogicalOperator{LogicalOperatorType::FILTER, std::move(child)},
-          expression{std::move(expression)} {}
+          expression{std::move(expression)} {
+        this->cardinality = cardinality;
```

Comment:
I don't think this the best way to set cardinality information in the constructor. Why don't we change LogicalOperator's constructor and make sure they take in cardinality as a constructor param?

Reply:
Updated, single-child constructor for LogicalOperator now takes in an optional cardinality param (if it is not provided it defaults to the cardinality of the child)